plan/plan.go

```diff
@@ -21,20 +21,58 @@ import (
 	"github.com/kubernetes-incubator/external-dns/endpoint"
 )
 
+// RecordKey is used to group records
+type RecordKey struct {
+	RecordType string
+	DNSName    string
+}
+
 // Plan can convert a list of desired and current records to a series of create,
 // update and delete actions.
 type Plan struct {
-	// List of current records
-	Current []*endpoint.Endpoint
-	// List of desired records
-	Desired []*endpoint.Endpoint
+	CurrentTargets map[RecordKey][]string
```

Comment:
It's a bit nit picky, but I would say the Current/Desired are still appropriate. The `CurrentTargets` made me do a double take on what the map actually represented. The RecordKey makes it more than just the Target